operator/crd/crd.yaml

```diff
@@ -168,28 +168,54 @@ spec:
                         - key
                         - type
                     keystore:
-                      description: "Configure HTTP server TLS connections using a JKS or PKCS12 keystore. The keystore password should be stored in a secret and referenced in the route Custom Resource. The format of the secret should be `password=<password>`."
+                      description: "Configures HTTP server TLS connections using a JKS or PKCS12 keystore. The keystore password should be stored in a Secret resource and referenced in the route's Custom Resource. The format of the Secret is `password=<password>`."
                       type: object
                       properties:
-                        secretName:
-                          type: string
-                        key:
-                          type: string
-                        type:
-                          type: string
-                          enum:
+                        jks:
+                          type: object
+                          properties:
+                            alias:
+                              description: "Specifies the unique identifier for the key in the JKS keystore. If not provided, the default alias of `certificate` is used."
+                              type: string
+                            secretName:
+                              description: "The name of the Secret resource containing the private key (tls.key), requsted certificate and certificate chain (tls.crt), keystore (keystore.jks), Certificate Authority's (CA) certificate (ca.crt), and truststore (truststore.jks)."
+                              type: string
+                            key:
+                              description: "The name of the key, containing the keystore, in the Secret resource (secretName)."
+                              type: string
+                            passwordSecretRef:
+                              description: "The reference to the Secret resource containing the password used to encrypt the JKS keystore."
+                              type: string
+                          required:
+                            - secretName
+                            - key
+                            - passwordSecretRef
+                        pkcs12:
+                          type: object
+                          properties:
+                            secretName:
+                              description: "The name of the Secret resource containing the private key (tls.key), requsted certificate and certificate chain (tls.crt), keystore (keystore.p12), Certificate Authority's (CA) certificate (ca.crt), and truststore (truststore.p12)."
```

Comment:
Same as comment above, seems only `keystore.p12` is relevant.

Reply:
Updated description